### PR TITLE
SWARM-1089: Quickstart using Relational Database

### PR DESCRIPTION
**Pull request by heiko-braun**
- database running in openshift
- rest interfaces extended to support CRUD operations
- web interface added that opeates on the rest interface
- database populated with example data
- readme instruction updated to reflect new workflow and purpose of the quickstart

### CODE REVIEW

**Comment by heiko-braun** [comment]
@cmoulliard @gastaldi @edewit anyone volunteering to do a quick review before merging?

**Comment by heiko-braun** [comment]
i.e. one question to me is: why does travis hook signal incompleteness, but when you look at the travis job everything is green?

File: src/main/java/org/obsidiantoaster/quickstart/service/FruitResource.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    @Produces("application/json")
+    @Transactional
+    public List search(@PathParam("token") String token) {
+        List<Fruit> fruits = em
+                .createNamedQuery("Fruits.findAll", Fruit.class)
+                .getResultList();
+
+        return fruits.stream()
+                .filter(f -> f.getName().contains(token))
+                .collect(Collectors.toList());
```
Comment (Ladicek) [inline src/main/java/org/obsidiantoaster/quickstart/service/FruitResource.java]: This isn't very real-world-ish :-)

File: src/main/java/org/obsidiantoaster/quickstart/service/FruitResource.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    @DELETE
+    @Path("/{id}")
+    @Consumes("text/plain")
+    @Transactional
+    public Response delete(@PathParam("id") Integer id) {
+        try {
+
+            Query query = em.createQuery("Select f from Fruit f where f.id=" + id);
```
Comment (Ladicek) [inline src/main/java/org/obsidiantoaster/quickstart/service/FruitResource.java]: This totally looks like "whoops are they really building database queries by concatenating strings?". I know `id` is an `Integer` (why not `int`?), so there's no risk, but we really should be using parameters even for such simple cases IMHO.

File: src/main/java/org/obsidiantoaster/quickstart/service/FruitResource.java (hunk 1 above)
Comment (heiko-braun) [reply src/main/java/org/obsidiantoaster/quickstart/service/FruitResource.java]: Do you think we should push it to the DB level?

File: src/main/java/org/obsidiantoaster/quickstart/service/FruitResource.java (hunk 2 above)
Comment (heiko-braun) [reply src/main/java/org/obsidiantoaster/quickstart/service/FruitResource.java]: you mean JPA query parameters?

File: src/main/java/org/obsidiantoaster/quickstart/service/FruitResource.java (hunk 1 above)
Comment (Ladicek) [reply src/main/java/org/obsidiantoaster/quickstart/service/FruitResource.java]: I'm not sure if I'd bother, it wouldn't add anything to the example. Personally, I'd just add a comment.

File: src/main/java/org/obsidiantoaster/quickstart/service/FruitResource.java (hunk 2 above)
Comment (Ladicek) [reply src/main/java/org/obsidiantoaster/quickstart/service/FruitResource.java]: Exactly.

**Comment by heiko-braun** [comment]
thanks @Ladicek. what you suggest makes sense. I'll update the samples

**Comment by gastaldi** [comment]
Looks good to me

**Pull request merged**